File: mobile/lib/services/authService.dart
import 'package:http/http.dart' as http;
import 'package:jwt_decoder/jwt_decoder.dart';
import 'dart:convert';
import 'package:shared_preferences/shared_preferences.dart';
import 'package:iRYDE/core/globals.dart';

class AuthService {
  Future<Map> login(Map user) async {
    try {
      final res = await http.post('$baseUrl/api/users/login',
          headers: null, body: user);
      SharedPreferences prefs = await SharedPreferences.getInstance();
      if (res.headers['token'] != null) {
        prefs.setString('token', res.headers['token']);
      }
      Map data = jsonDecode(res.body);
      return data;
    } catch (err) {
      print(err);
    }
  }

  Future<Map> currentUser() async {
    try {
      SharedPreferences prefs = await SharedPreferences.getInstance();
      String token = prefs.get('token');
      if (token != null) {
        Map<String, dynamic> decodedToken = JwtDecoder.decode(token);
        final res = await http.get(
          '$baseUrl/api/users/${decodedToken["_id"]}',
          headers: null,
        );
        Map data = jsonDecode(res.body);
        // print(data);
        return data;
      }
    } catch (err) {
      print(err);
    }
  }

  logout() async {
    SharedPreferences prefs = await SharedPreferences.getInstance();
    prefs.remove('token');
  }
}

File: web/src/services/authService.js
import axios from "axios";
import JwtDecode from "jwt-decode";

export async function login(data) {
  try {
    const user = await axios.post(
      "https://i-ryde-backend.herokuapp.com/api/users/login",
      data
    );
    localStorage.setItem("token", user.headers.token);
    // console.log(user.data);
    return user.data;
  } catch (ex) {
    console.log("exception", ex);
  }
}

export function logout() {
  localStorage.removeItem("token");
}

export async function getCurrentUser() {
  try {
    const userId = JwtDecode(localStorage.getItem("token"))._id;
    const currentUser = await axios.get(
      `https://i-ryde-backend.herokuapp.com/api/users/${userId}`
    );
    return currentUser;
  } catch (ex) {
    console.log("exception", ex);
  }
}

export default {
  login,
  logout,
  getCurrentUser,
};
